### ISSUE TITLE
Allow specifying the type of virtual keyboard to pop up in LineEdit and `OS.show_virtual_keyboard()`

### ISSUE THREAD

**Issue by Olaf-007** [godotengine/godot-proposals#1511]
**Describe the project you are working on:**
Hey I am currently working on an Android interface application.

**Describe the problem or limitation you are having in your project:**
I have the user inserting multiple names in a list of LineEdits and want to ease to jump from one edit to the next.
But therefore I need a NOTIFICATION from the virtual keyboard that currently does not exist (correct me if I am wrong), when the inserted name is "accepted".
Also there is the Control property "focus_next" which works on Windows with the tab key but not on Android.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
When OS.show_virtual_keyboard() is called on Andoird this opens up:
![current_keyboard_layout](https://user-images.githubusercontent.com/56698992/93065266-c895f280-f678-11ea-96d2-bfd539ec8d33.png)
But in other applications there are different kinds of keyboards which I would like to use in my application and cannot:
![otther_keyboard_options](https://user-images.githubusercontent.com/56698992/93065385-f2e7b000-f678-11ea-9d67-7708a7c43c29.png)

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
In Sketchware there are these two properties that can be set on a LineEdit or TextEdit:
![Bild1](https://user-images.githubusercontent.com/56698992/93065508-16125f80-f679-11ea-953d-700230001de9.png)
![Bild2](https://user-images.githubusercontent.com/56698992/93065517-1874b980-f679-11ea-9034-5bb645c6c166.png)
This could be a FLAG-property of the LineEdit or an optional FLAG for the OS.show_virtual_keyboard()-function:
OS.show_virtual_keyboard(type=OS.VIRTUAL_KEYBOARD_TYPE_TEXT,option=OS.VIRTUAL_KEYBOARD_OPTION_SEARCH)

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
I dont think so.

**Is there a reason why this should be core and not an add-on in the asset library?:**
In every other application in the app store this feature is used - from the calculator to any browser. Why is Godot not supporting this feature? (Correct me if I am wrong, but I did not found this setting where I expected it)

**Comment by xarses** [comment]
I too also have similar needs that would benefit from being able to better tell mobile about what type of virtual keyboards and was just heading over here to open a similar request, so I'll tack on here (let me know if you'd like it separate.



**Describe the project you are working on:**
Making a portable stream deck, which makes heavy use of GUI on mobile to perform automation during a broadcast.  

**Describe the problem or limitation you are having in your project:**
I went to make a number only input control, and have found the options limiting. Only `spinBox` supports only number entry, but requires the arrow buttons on the side of the control element, but offers no change the virtual keyboard

It appears that there is already some [terse](https://github.com/godotengine/godot/blob/master/platform/android/java/lib/src/org/godotengine/godot/input/GodotEditText.java#L124) setting of the Android virtual keyboard, I couldn't find any for iOS.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Both [Android](https://developer.android.com/reference/android/widget/TextView#attr_android:inputType) and [iOS](https://developer.apple.com/documentation/uikit/uikeyboardtype) already support typed virtual keyboards. Exposing the union of these would allow the developer to choose the most effective virtual keyboard type for the data they would like to capture

Minimally exposing the type to something like a `LineEdit` would allow for determined individuals to be able to usefully craft their own input fields as desired

Maximally supporting fully validated data entry types would be nice, but should be done in another proposal

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
In the simplest form, one would simply be able to specify the  virtual keyboard type to be used.

```gdscript
var InputField = LineEdit.new()
InputField.virtual_keyboard_type=LineEdit.VKBD_NUMBER
```

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
No, afaic core mechanics need to be added to be able perform this function in scrip as it's currently not plumbed. 

**Is there a reason why this should be core and not an add-on in the asset library?:**
Without adding and exposing the keyboard type to the underlying iOS and Android implementations, it will not be possible to generate mobile apps that can inform the OS about what type of virtual keyboard to draw on screen.

**Comment by brianwinterpixel** [comment]
I've got a working implementation for this in Godot 3.x. Hoping to put a PR together for `master` and `3.x` branches soon.

**Comment by akien-mga** [comment]
Implemented by https://github.com/godotengine/godot/pull/58536.

**Comment by Exoow** [comment]
@brianwinterpixel I just got Godot 3.5.1 and I don't find any property or method on LineEdit to set the virtual keyboard type. Sorry for noob question but how do I use this?

**Comment by Zireael07** [comment]
@Exoow It doesn't seem to have been backported to 3.x (it was only made in master, i.e. 4.x)

**Comment by brianwinterpixel** [comment]
> @brianwinterpixel I just got Godot 3.5.1 and I don't find any property or method on LineEdit to set the virtual keyboard type. Sorry for noob question but how do I use this?

It was merged too late for 3.5. You should see it in 3.6
